web/src/pages/index.tsx

```diff
@@ -271,306 +53,91 @@ const Home: NextPage = () => {
 
   // initial load
   useEffect(() => {
-    const mode = localStorage.getItem("chat_mode") as Mode || "default";
+    const mode = (localStorage.getItem("chat_mode") as Mode) || "default";
     setMode([mode, true]);
   }, []);
 
+  const updateCurrent = (current: CurrentSearch) => {
+    setCurrent(current);
+    if (current?.phase === "streaming") {
+      scroll30();
```

Comment:
this might not be needed, but I wanted to keep as close to the original code as possible